### DISCUSSION TITLE
SignIn callback returning a relative path fails

### DISCUSSION THREAD

**Discussion by nhuethmayr** [nextauthjs/next-auth#8025] [Help]
Hi guys,

I'm having the following piece of code in my project

```
signIn: async ({ user }) => {
  //signing in -> email must exist...
  const unknownUser = user.emailVerified === null

  if (unknownUser) {
    return new URL(
      '/sign-in?error=UnknownUser',
      process.env.NEXTAUTH_URL
    ).toString()
  }

  return true
}
```

As you can see, instead of returning a `relative` URL - I'm returning an absolute one.

This is due to an error I get when doing otherwise.
![image](https://github.com/nextauthjs/next-auth/assets/26460692/cf3334d5-48cf-455f-bb72-4eaa45663868)

From the browser
![image](https://github.com/nextauthjs/next-auth/assets/26460692/9bdd9ec2-06f6-4dfb-91aa-5bb215d0fb40)

and here the source
https://github.com/nextauthjs/next-auth/blob/1e886b97bcd76b0679a00d6d03e9cc0a68ca9407/packages/next-auth/src/react/index.tsx#L277

The error above can be avoided by returning an absolute URL which then causes problems on Vercel because the environment variable NEXTAUTH_URL does not exist.

https://next-auth.js.org/configuration/callbacks#sign-in-callback
The documentation clearly states that I should be able to return a relative path so I'm wondering what I need to do for this to work properly. I'd most certainly prefer a solution that does not require me to do string-concat magic

**Comment by nicoladj77** [comment]
@nhuethmayr did you ever fix this?

**Reply by nhuethmayr** [reply]
Let me see
